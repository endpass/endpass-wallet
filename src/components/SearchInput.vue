<template>
  <div class="search-control field is-expanded has-addons">
    <div class="control is-expanded has-icons-left">
      <input
        :value="value"
        class="input is-small"
        type="text"
        placeholder="search"
        @input="$emit('input', $event.target.value)"
      >
      <span
        class="icon is-small is-left"
        v-html="require('@/img/magnifying-glass.svg')"
      />
    </div>

    <div class="control">
      <a
        class="button is-small is-white"
        @click="$emit('input', '')"
      >
        <span
          class="icon is-small"
          v-html="require('@/img/x.svg')"
        />
      </a>
    </div>
  </div>
</template>

<script>
export default {
  name: 'SearchInput',
  props: {
    value: {
      type: String,
      default: '',
    },
  },
};
</script>
