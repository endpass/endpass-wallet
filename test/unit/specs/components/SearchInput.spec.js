import { shallow } from '@vue/test-utils';

import SearchInput from '@/components/SearchInput';

describe('SearchInput', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(SearchInput, {
      propsData: {
        value: '0',
      },
    });
  });

  describe('render', () => {
    it('should be a Vue component', () => {
      expect(wrapper.name()).toBe('SearchInput');
      expect(wrapper.isVueInstance()).toBeTruthy();
    });

    it('should render initial state of the component', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('behavior', () => {
    it('should emit event with entered value', () => {
      const value = '11';
      const input = wrapper.find('.input');
      input.element.value = value;
      wrapper.find('.input').trigger('input');
      expect(wrapper.emitted('input')).toBeTruthy();
      expect(wrapper.emitted().input[0]).toEqual([value]);
    });

    it('should emit event with empty value when clicked on clear', () => {
      const value = '11';
      wrapper.find('.button').trigger('click');
      expect(wrapper.emitted('input')).toBeTruthy();
      expect(wrapper.emitted().input[0]).toEqual(['']);
    });
  });
});
