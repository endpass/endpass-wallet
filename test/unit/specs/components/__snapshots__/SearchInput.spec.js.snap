// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput render should render initial state of the component 1`] = `
<div
  class="search-control field is-expanded has-addons"
>
  <div
    class="control is-expanded has-icons-left"
  >
    <input
      class="input is-small"
      placeholder="search"
      type="text"
    />
     
    <span
      class="icon is-small is-left"
    >
      {}
    </span>
  </div>
   
  <div
    class="control"
  >
    <a
      class="button is-small is-white"
    >
      <span
        class="icon is-small"
      >
        {}
      </span>
    </a>
  </div>
</div>
`;
